### ISSUE TITLE
Knowledge Box Detect Changes When No Changes Made

### ISSUE THREAD

**Issue by kvb2univpitt** [cmu-phil/tetrad#1365]
1. Set knowledge: explicitly added required and forbidden edges.
    ![add_knowledge](https://user-images.githubusercontent.com/10729309/154889276-56bad77b-c5f2-46e7-b144-183514672750.png)
2. Run FCI.
3. Reopen knowledge box.
    ![revisit_knowledge](https://user-images.githubusercontent.com/10729309/154889390-7c5456b2-8c55-44f9-b0be-94e25242694b.png) 
4. Check boxes to show required and forbidden edges explicitly.
    ![show_explicit_edges](https://user-images.githubusercontent.com/10729309/154889711-4af73adf-d55b-4f19-8950-91411943be89.png)
5. Click "Done" and a warning message shows up stating that changes will affect its children.
    ![knowledge_change_warning](https://user-images.githubusercontent.com/10729309/154889652-e320b1b0-4948-4b11-a356-d1978c3230b3.png)

Open up the knowledge box and repeat this few times.  The knowledge box eventually breaks.
![knowledge_error](https://user-images.githubusercontent.com/10729309/154889817-80cee10a-8b67-49e1-8921-fb34139f290e.png)

**Comment by jdramsey** [comment]
@kvb2univpitt I'm having trouble reproducing this. Is it a bug in tetrad-fixes-joe-try2?

**Comment by kvb2univpitt** [comment]
Yes, it is.  I figured the problem.   Every time **Show Required Explicitly** button is clicked, the same edges gets added to the list.

I printed out the edge list:
```
Required Edges
X5-->X8
X5-->X4
X5-->X8
X5-->X4
X5-->X8
X5-->X4
X5-->X8
X5-->X4
Forbidden Edges
X1-->X2
X1-->X6
```
As you can see, I clicked on the checkbox 4 times I get 8 items in the list.  Each list should only contains 2 items.

**Comment by jdramsey** [comment]
OK I think I finally fixed this.
